### PR TITLE
Apexmalhar 2013 hdfs file copy module

### PR DESCRIPTION
**Pull request by yogidevendra**

### CODE REVIEW

File: apps/filecopy/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/target/
```
Comment (DT-Priyanka) [inline apps/filecopy/.gitignore:1]: You might want to add .classpath, .project, .settings/, .metadata/, *.iml, .idea/ as well.

File: apps/filecopy/pom.xml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>malhar-apps</artifactId>
+    <groupId>org.apache.apex</groupId>
+    <version>3.4.0-incubating-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.apache.apex</groupId>
```
Comment (DT-Priyanka) [inline apps/filecopy/pom.xml:30]: You can omit this, groupId would be picked up from parent.

File: apps/filecopy/pom.xml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  <artifactId>filecopy</artifactId>
+  <packaging>jar</packaging>
+
+  <properties>
+    <maven.deploy.skip>true</maven.deploy.skip>
+    <skipTests>false</skipTests>
+    <semver.plugin.skip>true</semver.plugin.skip>
+  </properties>
+
+  <name>Apache Apex Malhar (incubating) Logstream Application</name>
```
Comment (DT-Priyanka) [inline apps/filecopy/pom.xml:40]: Please change the name.

File: apps/filecopy/pom.xml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+  <build>
+    <plugins>
+      <plugin>
+        <artifactId>maven-eclipse-plugin</artifactId>
+      </plugin>
+      <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
```
Comment (DT-Priyanka) [inline apps/filecopy/pom.xml:48]: I am not sure, but don't you need to provide any configurations for these plugins? May be it refers parent, just need to confirm how it works.

File: apps/filecopy/src/assemble/appPackage.xml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+-->
+<assembly xmlns="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.2"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.2 http://maven.apache.org/xsd/assembly-1.1.2.xsd">
+  <id>appPackage</id>
+  <formats>
+    <format>jar</format>
+  </formats>
+  <includeBaseDirectory>false</includeBaseDirectory>
+  <!--files>
```
Comment (DT-Priyanka) [inline apps/filecopy/src/assemble/appPackage.xml:29]: remove commented code.

File: apps/filecopy/src/main/java/com/datatorrent/modules/app/HDFSFileCopySampleApp.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.datatorrent.modules.app;
```
Comment (DT-Priyanka) [inline apps/filecopy/src/main/java/com/datatorrent/modules/app/HDFSFileCopySampleApp.java:20]: may be com.datatorrent.apps.copy
Why are you having modules in package name?

File: apps/filecopy/src/main/java/com/datatorrent/modules/app/HDFSFileCopySampleApp.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.DAG.Locality;
+import com.datatorrent.api.StreamingApplication;
+import com.datatorrent.api.annotation.ApplicationAnnotation;
+import com.datatorrent.lib.io.fs.HDFSFileCopyModule;
+import com.datatorrent.lib.io.fs.HDFSInputModule;
+
+/**
+ * Sample application for HDFS to HDFS file copy
```
Comment (DT-Priyanka) [inline apps/filecopy/src/main/java/com/datatorrent/modules/app/HDFSFileCopySampleApp.java:32]: Is this going to be a sample app for reference, or a consumable application?

**Comment by yogidevendra** [comment]
Closing this PR as commit logs are not shown as per expectation.

**Comment by yogidevendra** [comment]
New PR is here : https://github.com/apache/incubator-apex-malhar/pull/216/files
I will also consider comments mentioned above while incorporating changes on new PR.

File: apps/filecopy/pom.xml [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+      <artifactId>malhar-library</artifactId>
+      <version>3.4.0-incubating-SNAPSHOT</version>
+      <exclusions>
+        <exclusion>
+          <groupId>*</groupId>
+          <artifactId>*</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+  <!--
```
Comment (DT-Priyanka) [inline apps/filecopy/pom.xml:91]: Please remove commented code.

File: apps/filecopy/src/site/conf/app-conf.xml [hunk 9]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<configuration>
+  <property>
+    <name>dt.operator.HDFSInputModule.prop.files</name>
+    <value>hdfs://localhost:54310/user/dtadmin/input</value>
+  </property>
+  <property>
+    <name>dt.operator.HDFSInputModule.prop.blockSize</name>
```
Comment (DT-Priyanka) [inline apps/filecopy/src/site/conf/app-conf.xml:8]: This size is in mb right? Why such big value?

**Comment by tweise** [comment]
Please get rid of the history of unrelated commits.

File: library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileSplitter.java [hunk 10]
```diff
@@ -365,6 +365,16 @@ public FileMetadata(@NotNull String filePath)
       this.filePath = filePath;
       discoverTime = System.currentTimeMillis();
     }
+    
+    protected FileMetadata(FileMetadata fileMetadata)
+    {
+      this();
+      this.filePath = fileMetadata.filePath;
```
Comment (DT-Priyanka) [inline library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileSplitter.java:372]: minor thing, no need of using "this" here.

File: apps/filecopy/.gitignore (hunk 1 above)
Comment (yogidevendra) [inline apps/filecopy/.gitignore:1]: Removed this file. parent .gitignore will serve the purpose.

File: apps/filecopy/pom.xml (hunk 2 above)
Comment (yogidevendra) [inline apps/filecopy/pom.xml:30]: removed.

File: apps/filecopy/pom.xml (hunk 3 above)
Comment (yogidevendra) [inline apps/filecopy/pom.xml:40]: name changed

File: apps/filecopy/pom.xml (hunk 4 above)
Comment (yogidevendra) [inline apps/filecopy/pom.xml:48]: It refers parent.

File: apps/filecopy/pom.xml (hunk 8 above)
Comment (yogidevendra) [inline apps/filecopy/pom.xml:91]: removed.

File: apps/filecopy/src/assemble/appPackage.xml (hunk 5 above)
Comment (yogidevendra) [inline apps/filecopy/src/assemble/appPackage.xml:29]: removed.

File: apps/filecopy/src/main/java/com/datatorrent/modules/app/HDFSFileCopySampleApp.java (hunk 6 above)
Comment (yogidevendra) [inline apps/filecopy/src/main/java/com/datatorrent/modules/app/HDFSFileCopySampleApp.java:20]: changed package declaration.

File: apps/filecopy/src/main/java/com/datatorrent/modules/app/HDFSFileCopySampleApp.java (hunk 7 above)
Comment (yogidevendra) [inline apps/filecopy/src/main/java/com/datatorrent/modules/app/HDFSFileCopySampleApp.java:32]: Changed the name to HDFSFileCopyApp

File: apps/filecopy/src/site/conf/app-conf.xml (hunk 9 above)
Comment (yogidevendra) [inline apps/filecopy/src/site/conf/app-conf.xml:8]: Looking at the code blockSize is in bytes. @DT-Priyanka Please confirm.

File: apps/filecopy/src/site/conf/app-conf.xml [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  </property>
+  <property>
+    <name>dt.operator.HDFSFileCopyModule.prop.outputDirectoryPath</name>
+    <value>hdfs://localhost:54310/user/dtadmin/output</value>
+  </property>
+  <property>
+    <name>dt.loggers.level</name>
+    <value>com.datatorrent.*:DEBUG,org.apache.*:INFO</value>
+</property>
+</configuration>
```
Comment (yogidevendra) [inline apps/filecopy/src/site/conf/app-conf.xml:27]: Added new line at the end.

File: library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileSplitter.java (hunk 10 above)
Comment (yogidevendra) [inline library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileSplitter.java:372]: removed this references.

Comment (yogidevendra) [inline library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileSplitter.java:372]: remove 'this' reference.

**Pull request closed without merging**